wls-wahlvorbereitung-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/security/AuthUtils.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c): it@M - Dienstleister für Informations- und Telekommunikationstechnik
+ * der Landeshauptstadt München, 2024
+ */
+package de.muenchen.oss.wahllokalsystem.wahlvorbereitungservice.security;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+
+/**
+ * Utilities zu Authentifizierungsdaten.
+ *
+ * @author michael.prankl
+ *
+ */
+public class AuthUtils {
+
+    public static final String NAME_UNAUTHENTICATED_USER = "unauthenticated";
+
+    private static final String TOKEN_USER_NAME = "user_name";
+
+    private AuthUtils() {
+    }
+
+    /**
+     * Extrahiert den Usernamen aus dem vorliegenden Spring Security Context via
+     * {@link SecurityContextHolder}.
+     *
+     * @return der Username or a "unauthenticated", wenn keine {@link Authentication} existiert
+     */
+    public static String getUsername() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication instanceof JwtAuthenticationToken) {
+            JwtAuthenticationToken jwtAuth = (JwtAuthenticationToken) authentication;
+            return (String) jwtAuth.getTokenAttributes().getOrDefault(TOKEN_USER_NAME, null);
+        } else if (authentication instanceof UsernamePasswordAuthenticationToken) {
+            UsernamePasswordAuthenticationToken usernameAuth = (UsernamePasswordAuthenticationToken) authentication;
+            return usernameAuth.getName();
+        } else {
+            return NAME_UNAUTHENTICATED_USER;
+        }
```

Comment:
Gerne deinen Vorschlag zum Patternmatching auch hier verwenden (https://github.com/it-at-m/Wahllokalsystem/pull/200/commits/d53185be88bb05a6d20f886d37a9348454564d15). Bei `UserInfoAuthoritiesService` hattest du es bereits angewendet.